libres/lib/enkf/local_updatestep.cpp

```diff
@@ -31,55 +30,35 @@
    contain several.
 */
 
-#define LOCAL_UPDATESTEP_TYPE_ID 77159
-
-struct local_updatestep_struct {
-    UTIL_TYPE_ID_DECLARATION;
-    char *name;
-    vector_type *ministep;
-};
-
-UTIL_SAFE_CAST_FUNCTION(local_updatestep, LOCAL_UPDATESTEP_TYPE_ID)
+LocalUpdateStep::LocalUpdateStep(const std::string &name) : m_name(name) {}
 
-local_updatestep_type *local_updatestep_alloc(const char *name) {
-    local_updatestep_type *updatestep =
-        (local_updatestep_type *)util_malloc(sizeof *updatestep);
+std::size_t LocalUpdateStep::size() const { return this->m_ministep.size(); }
 
-    UTIL_TYPE_ID_INIT(updatestep, LOCAL_UPDATESTEP_TYPE_ID);
-    updatestep->name = util_alloc_string_copy(name);
-    updatestep->ministep = vector_alloc_new();
+const std::string &LocalUpdateStep::name() const { return this->m_name; }
 
-    return updatestep;
+LocalMinistep &LocalUpdateStep::operator[](std::size_t index) {
+    return this->m_ministep[index].get();
 }
 
-void local_updatestep_free(local_updatestep_type *updatestep) {
-    free(updatestep->name);
-    vector_free(updatestep->ministep);
-    free(updatestep);
+const LocalMinistep &LocalUpdateStep::operator[](std::size_t index) const {
+    return this->m_ministep[index].get();
 }
 
-void local_updatestep_free__(void *arg) {
-    local_updatestep_type *updatestep = local_updatestep_safe_cast(arg);
-    local_updatestep_free(updatestep);
+void LocalUpdateStep::add_ministep(LocalMinistep &ministep) {
+    this->m_ministep.push_back(std::ref(ministep));
```

Comment:
Well - because the ministep vector is `std::vector<std::reference_wrapper<LocalMinistep>>`  - and that again is minimal change from the original implementation. Briefly described in PR description.